### ISSUE TITLE
Nextcloud Keeweb config not saved

### ISSUE THREAD

**Issue by gabsor** [keeweb/keeweb#1175]
**Describe the bug**
I'm using Keeweb app in Nextcloud on Synology. When changing Keeweb general settings like "Language" or  "Theme" or any other setting, these are not saved. Next login all is reset to default. 
Also tried to load configuration settions as described here: https://github.com/keeweb/keeweb/wiki/Configuration#json-app-config.
Cleared browser cache, but settings are not loaded. 
So bug is: configuration settings are not saved

Same applies to the plugins, they get installed but are gone after logout of Nextcloud. 

**To Reproduce**
Go to Synology / Nextcloud / Keeweb app, load *.kdbx, open general settings page.
Change settings. Log-out. Log-in again, see that settings are reset to default e.g. not saved. 

**Expected behavior**
Would expect the settings to be saved under the Nextcloud user that is opening /using Keeweb, or saving the new settings as the new global default for all users. I do not expect the settings to return to the default as set during install.
Would also expect that plugins remain installed and operational after a logout. 

Alternative: indicate where and how to change the global general config settings

**Environment**
Your user-agent (from Settings/Help section)
KeeWeb v1.7.7 (4ca2eb4c, 2019-02-11)
Environment: web
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:66.0) Gecko/20100101 Firefox/66.0

and

KeeWeb v1.7.7 (4ca2eb4c, 2019-02-11)
Environment: web
User-Agent: Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.86 YaBrowser/19.4.0.2397 Yowser/2.5 Safari/537.36

**Kdbx File**
Does it happen on Demo or New database?  
If you have a test db without your personal data, please attach it.

It happens to all databases. 

**Logs**
Please open dev tools in your browser and attach output log from Console tab. If you are using a desktop app, devtools can be opened from Settings/General/Advanced.

Not log related e.g. no errors in log related to this.

**Comment by antelle** [comment]
Hi! Settings are saved in browser local storage, could you please check what's there (it's available in dev tools)?
Maybe you're using a browser in private mode? Then it won't be possible to save them.

**Comment by gabsor** [comment]
Hi, not entirely sure what you mean. Using Yandex and Firefox. No chrome ;( 
See the attached prtscn... hope it helps. I do clear browser cache after usage (closing browser).
Tx in advance!
![dev](https://user-images.githubusercontent.com/10881902/57331227-5d4d1f80-7118-11e9-9fbf-ba149a3ddae7.jpg)

**Comment by antelle** [comment]
What happens if you don't clear browser cache, do settings persist?

**Comment by gabsor** [comment]
Just tested that with Firefox (latest), changed settings e.g. not to delete history / clear cache. I logged in, changed confi settings and installed / configured another language (NL)).  Closed browser. Opened browser, logged in and opened Keeweb. Can already see on the main page that results are resetted since languate = English. After login, all config settings resetted to its original.

**Comment by antelle** [comment]
Weird. Looks like localStorage is cleared for some reason, not sure why this can happen. What happens if you type this in devtools:
```
localStorage.test = 1
```
then open keeweb once again as you do, and type
```
localStorage.test
```
It should give 1. If it doesn't, that's a problem.

**Comment by gabsor** [comment]
Btw in session storage / buffer storage / local storage (fire fox ctrl-shif-i) I do not see much data stored.

**Comment by gabsor** [comment]
Maybe its because I load the *.kdbx file from Nextcloud (mime type association with Keeweb) automatically it looses its config settings? I see in the firefox console:  Loading config from config?file=/shared/xxxx_yyyyy.kdbx . Does the *.kdbx contain the Keeweb config of am I mixing up things?

**Comment by antelle** [comment]
> Loading config from config file

Right, that's it, thanks. The config file overrides settings you changed.

**Comment by gabsor** [comment]
So I should adjust settings in f.e. Keepass and use that *.kdbx file in Keeweb?

**Comment by antelle** [comment]
No, settings are not saved in the kdbx file. A config file is another thing, which contains your settings and it overrides what you change. If you would like to be able to change something, you should remove it from the config.

**Comment by gabsor** [comment]
Hmmm, but if I indicate a config file to use with https://www.domain.eu/nexcloud/index.php/apps/keeweb/config?file=config.json
that config is not loaded / read. I always get the same default config settings. 
So how can I change those without having to set it every session over and over again?

**Comment by antelle** [comment]
I'm not sure if I understand. What do you have in your config? For example, if you would like to be able to change the interface language, you should remove it from the config.

**Comment by gabsor** [comment]
For example, this config.json: 
{
"settings": {
"theme": "fb",
"locale": "nl",
"expandGroups": true,
"autoUpdate": "check",
"autoSave": true,
"rememberKeyFiles": false,
"idleMinutes": 60,
"tableView": false,
"colorfulIcons": true,
"lockOnCopy": false,
"skipOpenLocalWarn": false,
"hideEmptyFields": true,
"skipHttpsWarning": false,
"demoOpened": false,
"fontSize": 1,
"tableViewColumns": null,
"generatorPresets": null,
"cacheConfigSettings": false,
"canOpen": true,
"canOpenDemo": false,
"canOpenSettings": true,
"canCreate": true,
"canImportXml": true,
"dropbox": false,
"dropboxFolder": null,
"dropboxAppKey": null,
"webdav": true,
"webdavSaveMethod": "put",
"gdrive": false,
"gdriveClientId": null,
"onedrive": false,
},
"files": [{
"path": "xxxx_yyyy.kdbx"
"storage": "webdav",
"name": "GABSTOR",
"path": "webdav-url",
"options": { "user": "", "password": "" }
}],
}

But again, its not loaded. It resides in the docroot of my webserver. 

Maybe a feature request, in case a local / server config.json exists, load it as default?

**Comment by antelle** [comment]
I'm confused even more now. Earlier you said that the config is loaded. So, what happens for you: is the config loaded or not?

**Comment by gabsor** [comment]
I said that in the firefox dev console I saw this line netry stating: Loading config from config?file=/shared/xxxx_yyyyy.kdbx... but you said that's a *.kdbx file not containing the config settings like language, etc.  So to be clear: I open a *.kdbx from nextcloud (mime type association to open *.kdbx files with the keepass app)... thats working fine. But, loading that file with the preferred / personal configuration settings is not working / loaded. They are changed within the browser session when I go to settings and change items.. but after logging out / closing the browser they are set back to their original setting. What I want is to use the config settings that I define without having to set them everytime I start using Keeweb. Excellent app, really like it but this is the only issue (users that not spreak English and need Dutch).... Clear?

**Comment by antelle** [comment]
For sure, I understand the need of saving settings, you don't have to stress this. 😉 
What's not clear is why this happens and it takes already too long to debug this, sorry, I can't afford it. I'll dump some facts that I have in mind and that can help you in investigation:

- kdbx files never contain settings
- settings are stored in [config.json](https://github.com/keeweb/keeweb/wiki/Configuration#json-app-config)
- after loading config.json settings are saved to localStorage
- when you change setting in the UI, they're also saved to localStorage
- when the config is loaded, existing settings can be overridden
- localStorage must work (as in, be persistent) to be able to use settings
- [nextcloud-keeweb](https://github.com/jhass/nextcloud-keeweb) is a separate project
- there were no other such reports so far

**Comment by gabsor** [comment]
No stress, just trying to be clear ;) Can imagine it is sometimes hard to follow. 
Thanks so far, will try to investigate and fix and see if nextcloud-keeweb experiences with this exist. 
In case I have news, I will share it.  Regards!
